src/components/spoiler/spoiler.js

```diff
@@ -0,0 +1,35 @@
+import Component from '../../app/js/base/Component';
+
+class Spoiler extends Component {
+    iconElement;
+    textElement;
+    isOpen = true;
+
+    constructor(element) {
+        super(element);
+
+        this.textElement = this.getElement('text')
+        this.iconElement = this.getElement('svg')
+
+        this.root.addEventListener('click', this.onButtonClick)
+    }
+
+    onButtonClick = () => {
+
+        if (this.isOpen){
+            const scrollHeight = this.textElement.scrollHeight
+            this.textElement.classList.toggle('open')
+            this.iconElement.classList.toggle('open')
+            this.textElement.style.height = `${scrollHeight}px`
+            this.isOpen = !this.isOpen
+        }else{
+            this.textElement.classList.toggle('open')
+            this.iconElement.classList.toggle('open')
+            this.textElement.style.height = '0'
+            this.isOpen = !this.isOpen
+        }
+
+    }
```

Comment:
Можно вынести действия по закрытию/открытию в отдельные методы класса, чтобы упростить чтение кода. Это также позволит использовать методы open/close за пределами обработчика события
```suggestion
    onButtonClick = () => {
        this.isOpen ? this.close() : this.open();
    }
    
    open() {
        this.textElement.classList.add('open');
        this.iconElement.classList.add('open');
        this.textElement.style.height = `${this.textElement.scrollHeight}px`;
        this.isOpen = true;
    }
    
    close() {
        this.textElement.classList.remove('open');
        this.iconElement.classList.remove('open');
        this.textElement.style.height = '0';
        this.isOpen = false;
    }
```